# **A) Average Order value gives a way to know how sustainable a business is but however it does not really help in this case because of the price margin of the sneakers. For example, shop_id 78 has an order amount of 25725 per item which is the maximum while shop_id 92 has an order amount of 90 per item.  AOV works best when there is a tight product price range.**

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")

In [3]:
data.head()

order_id  shop_id  user_id  ...  total_items  payment_method           created_at
0         1       53      746  ...            2            cash  2017-03-13 12:36:56
1         2       92      925  ...            1            cash  2017-03-03 17:38:52
2         3       44      861  ...            1            cash   2017-03-14 4:23:56
3         4       18      935  ...            1     credit_card  2017-03-26 12:43:37
4         5       18      883  ...            1     credit_card   2017-03-01 4:35:11

[5 rows x 7 columns]

In [4]:
x = sum(data['order_amount'])

In [8]:
z = len(data['order_id'])

In [9]:
x/z

3145.128

In [10]:
len(data['user_id'].unique())

301

In [18]:
data['amount per unit'] = data['order_amount']//data['total_items']

In [19]:
data.head()

order_id  shop_id  ...           created_at  amount per unit
0         1       53  ...  2017-03-13 12:36:56              112
1         2       92  ...  2017-03-03 17:38:52               90
2         3       44  ...   2017-03-14 4:23:56              144
3         4       18  ...  2017-03-26 12:43:37              156
4         5       18  ...   2017-03-01 4:35:11              156

[5 rows x 8 columns]

In [20]:
max(data['amount per unit'])

25725

In [21]:
min(data['amount per unit'])

90

In [31]:
max_data = data[data['amount per unit'] == max(data['amount per unit'])]

In [32]:
max_data.head()

order_id  shop_id  ...           created_at  amount per unit
160       161       78  ...   2017-03-12 5:56:57            25725
490       491       78  ...  2017-03-26 17:08:19            25725
493       494       78  ...  2017-03-16 21:39:35            25725
511       512       78  ...   2017-03-09 7:23:14            25725
617       618       78  ...  2017-03-18 11:18:42            25725

[5 rows x 8 columns]

In [33]:
max_data['shop_id'].unique()

array([78])

In [34]:
min_data = data[data['amount per unit'] == min(data['amount per unit'])]

In [35]:
min_data.head()

order_id  shop_id  ...           created_at  amount per unit
1           2       92  ...  2017-03-03 17:38:52               90
97         98       92  ...  2017-03-22 19:21:26               90
158       159       92  ...   2017-03-29 3:07:12               90
228       229       92  ...  2017-03-13 23:57:51               90
236       237       92  ...  2017-03-18 20:51:12               90

[5 rows x 8 columns]

In [36]:
min_data['shop_id'].unique()

array([92])

# **B)Purchase Frequenct is agreat KPI to measure customer loyalty and it is the average number of orders customers made during the 30-day window**

In [43]:
Purchase_frequency = (len(data['order_id']) / (len(data['user_id'].unique())))

In [44]:
Purchase_frequency

16.611295681063122

# **C) The value is 16.6**

# **Another metric is the Repeat Purchase Rate (RPR) which tells the percentage of returning customers. In this case all the customers have visited and returned back to the store more than once.**

In [45]:
data.head()

order_id  shop_id  ...           created_at  amount per unit
0         1       53  ...  2017-03-13 12:36:56              112
1         2       92  ...  2017-03-03 17:38:52               90
2         3       44  ...   2017-03-14 4:23:56              144
3         4       18  ...  2017-03-26 12:43:37              156
4         5       18  ...   2017-03-01 4:35:11              156

[5 rows x 8 columns]

In [49]:
len(data['user_id'].value_counts() > 2)

301

In [50]:
len(data['user_id'].unique())

301

In [51]:
len(data['user_id'].value_counts() > 2) / len(data['user_id'].unique())

1.0